hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -784,22 +785,6 @@ object DataSourceOptionsHelper {
     ) ++ translateConfigurations(paramsWithGlobalProps)
   }
 
-  def inferKeyGenClazz(props: TypedProperties): String = {
```

Comment:
Move this logic to `HoodieSparkKeyGeneratorFactory` as `HoodieDatasetBulkInsertHelper` need to decide `KeyGenClass`